src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUnary.cs

```diff
@@ -403,21 +403,8 @@ public static DateTimeValue TimeToDateTime(IRContext irContext, TimeValue[] args
 
         public static FormulaValue OptionSetValueToString(IRContext irContext, OptionSetValue[] args)
         {
-            // The type checker and IR have already validated that this is a valid OptionSet and that it contains a member matching this Option.
-            // These are just defensive error checks, and should be unreachable.
-            var os = args[0].Type?._type?.OptionSetInfo;
-            if (os is not OptionSet optionSet) 
-            {
-                return CommonErrors.UnreachableCodeError(irContext);
-            }
-
-            var option = args[0].Option;
-
-            if (!optionSet.Options.TryGetValue(new DName(option), out var displayName))
-            { 
-                return CommonErrors.UnreachableCodeError(irContext);
```

Comment:
>CommonErrors.UnreachableCodeError(irContext); [](http://example.com/codeflow?start=23&length=45)

Just made this return logical name when display name is missing.  rather than error.